client_src/js/services/filament-manager-plugin.service.js

```diff
@@ -0,0 +1,127 @@
+  const disableFilManagerBtn = document.getElementById("disable-FilamentManager");
+  disableFilManagerBtn.addEventListener("click", async (event) => {
+    let filamentManagerDisabled = await OctoFarmClient.post("filament/disableFilamentPlugin", {
+      activate: true
+    });
+    if (filamentManagerDisabled) {
+      UI.createAlert(
+        "success",
+        "Successfully disabled filament manager and removed all spools / profiles.",
+        3000
+      );
+    } else {
+      UI.createAlert(
+        "error",
+        "Unable to disable filament manager please check the filament manager logs.",
+        3000
+      );
+    }
+  });
+}
+
+export function setupFilamentManagerSyncBtn() {
+  const filamentManagerSyncBtn = document.getElementById("resyncFilamentManagerBtn");
+  filamentManagerSyncBtn.addEventListener("click", async (event) => {
+    UI.addLoaderToElementsInnerHTML(filamentManagerSyncBtn);
+    filamentManagerSyncBtn.disabled = true;
+    const filamentManagerSyncEnabled = await OctoFarmClient.post("filament/filamentManagerSync", {
+      activate: true
+    });
+    if (filamentManagerSyncEnabled) {
+      UI.createAlert(
+        "success",
+        "Successfully disabled filament manager and removed all spools / profiles.",
+        3000
+      );
+    } else {
+      UI.createAlert(
+        "error",
+        "Unable to disable filament manager please check the filament manager logs.",
+        3000
+      );
+    }
+    UI.removeLoaderFromElementInnerHTML(filamentManagerSyncBtn);
+    filamentManagerSyncBtn.disabled = false;
+  });
+}
+
+export function setupFilamentManagerReSyncBtn() {
+  const filamentManagerEnabled = isFilamentManagerPluginSyncEnabled();
+  if (filamentManagerEnabled) {
+    const resyncBtn = document.getElementById("resyncFilamentManagerBtn");
+    resyncBtn.addEventListener("click", async (e) => {
+      UI.addLoaderToElementsInnerHTML(resyncBtn);
+      const post = await OctoFarmClient.post("filament/filamentManagerReSync");
+      UI.createAlert(
+        "success",
+        `Successfully synced filament manager! <br> Profiles - Updated: ${post.updatedProfiles} / New: ${post.newProfiles} <br> Spools - Updated: ${post.updatedSpools} / New: ${post.newSpools}`,
+        4000,
+        "Clicked"
+      );
+      UI.removeLoaderFromElementInnerHTML(resyncBtn);
+    });
+  }
+}
+
+// export async function returnSelected(id, profiles) {
+//   console.log("Return Selected");
+//   let profileId = null;
+//   const filamentManager = await checkFilamentManager();
+//   if (filamentManager) {
+//     profileId = _.findIndex(profiles, function (o) {
+//       return o.profile.index == id.spools.profile;
+//     });
+//   } else {
+//     profileId = _.findIndex(profiles, function (o) {
+//       return o._id == id.spools.profile;
+//     });
+//   }
+//   return `${id.spools.name} (${(id.spools.weight - id.spools.used).toFixed(
+//     0
+//   )}g) - ${profiles[profileId].profile.material}`;
+// }
```

Comment:
I've left this there because at some point I wanna enable this back on filament manager

Reply:
A comment above it would be fine and helps you remember being your future self